x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/AuthenticateRequest.java

```diff
@@ -6,50 +6,38 @@
  */
 package org.elasticsearch.xpack.core.security.action.user;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.action.ActionRequest;
 import org.elasticsearch.action.ActionRequestValidationException;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 
 import java.io.IOException;
 
-public class AuthenticateRequest extends ActionRequest implements UserRequest {
+public class AuthenticateRequest extends ActionRequest {
 
-    private String username;
+    public static final AuthenticateRequest INSTANCE = new AuthenticateRequest();
 
     public AuthenticateRequest(StreamInput in) throws IOException {
         super(in);
-        username = in.readString();
+        if (in.getVersion().before(Version.V_8_4_0)) {
+            in.readString();
+        }
     }
 
-    public AuthenticateRequest() {}
-
-    public AuthenticateRequest(String username) {
-        this.username = username;
-    }
+    private AuthenticateRequest() {}
 
     @Override
     public ActionRequestValidationException validate() {
         // we cannot apply our validation rules here as an authenticate request could be for an LDAP user that doesn't fit our restrictions
         return null;
     }
 
-    public String username() {
-        return username;
-    }
-
-    public void username(String username) {
-        this.username = username;
-    }
-
-    @Override
-    public String[] usernames() {
-        return new String[] { username };
-    }
-
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        out.writeString(username);
+        if (out.getVersion().before(Version.V_8_4_0)) {
+            throw new IllegalStateException("cannot send authenticate request to a node of earlier version");
+        }
```

Comment:
Throwing exception here since it is better to fail explicitly than guessing. But in practice this will never happen because AuthenticateRequest does not get sent across nodes.